services/payments/statemachine_solana.go

```diff
@@ -198,16 +198,33 @@ func (sm *SolanaMachine) Pay(ctx context.Context) (*paymentLib.AuthenticatedPaym
 	}
 
 	b58Signature := base58.Encode(idempotencyData.Transaction.Signatures[0])
+	// solanaError stores temporary errors that need to be recorded as the LastError in QLDB but
+	// that are expected and do not need to be acted upon. It will only be returned if no other
+	// unexpected errors arise from processing.
+	var solanaError error
 	status, err := checkStatus(ctx, b58Signature, &sm.solanaRpcClient)
 	if err != nil {
-		return sm.transaction, fmt.Errorf("failed to check transaction status: %w", err)
+		// Some errors are expected and we just want to record them. Check if the status check
+		// returned such an error and prepare to return it after we attempt to make progress.
+		var (
+			errNotConfirmed SolanaTransactionNotConfirmedError
+			errNotFound     SolanaTransactionNotFoundError
+			errUnknown      SolanaTransactionUnknownError
```

Comment:
All your errors here should be pointers, because that's what implements the `error` interface.

```suggestion
			errNotConfirmed *SolanaTransactionNotConfirmedError
			errNotFound     *SolanaTransactionNotFoundError
			errUnknown      *SolanaTransactionUnknownError
```